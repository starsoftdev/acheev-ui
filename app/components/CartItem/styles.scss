@import '~styles/settings/settings';

.cartItem {
  color: $darker-gray;
  &:not(:last-child) {
    padding-bottom: $rem-xl;
    border-bottom: 1px solid $light-gray;
  }
  &__title {
    font-size: $fs-xl;
  }
  &__quantityLabel {
    padding-right: $rem-md;
  }
  &__quantityValue {
    padding: 0px $rem-sm;
  }
  &__quantityChange {
    border: 1px solid $darker-gray;
    width: $rem-mn;
    height: $rem-mn;
    box-sizing: border-box;
    padding: 0;
    border-radius: $rem-xxl;
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: center;
    cursor: pointer;
  }
  &__price {
    font-size: $fs-mx;
    font-weight: light;
  }
}
