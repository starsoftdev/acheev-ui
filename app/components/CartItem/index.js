// @flow

import * as React from 'react';

import './styles.scss';

type Props = {
  lineItem: Object,
  decreaseQuantity: Function,
  increaseQuantity: Function,
};

const CartItem = ({ lineItem, increaseQuantity, decreaseQuantity }: Props) => {
  if (!lineItem) return null;
  return (
    <div className="cartItem row mb-xl">
      <div className="small-3 column npl">
        <img
          className="cartItem__img"
          src={lineItem.image.src}
          alt={lineItem.title}
        />
      </div>
      <div className="small-9 column">
        <div className="cartItem__title mb-mn">{lineItem.title}</div>
        {lineItem.variant_title !== 'Default Title' && (
          <div className="cartItem__varitantTitle mb-mn">
            {lineItem.variant_title}
          </div>
        )}
        <div className="row align-middle mb-md fs-tn">
          <div className="cartItem__quantityLabel shrink column">QTY</div>
          <button
            className="cartItem__quantityChange shrink column"
            onClick={() => decreaseQuantity(lineItem)}
          >
            <span>-</span>
          </button>
          <div className="cartItem__quantityValue shrink column">
            {lineItem.quantity}
          </div>
          <button
            className="cartItem__quantityChange shrink column"
            onClick={() => increaseQuantity(lineItem)}
          >
            <span>+</span>
          </button>
        </div>
        <div className="cartItem__price">${lineItem.line_price}</div>
      </div>
    </div>
  );
};

export default CartItem;
