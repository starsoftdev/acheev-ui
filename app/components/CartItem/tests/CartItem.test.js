import React from 'react';
import { shallow, mount } from 'enzyme';
import { create } from 'react-test-renderer';

import CartItem from 'components/CartItem';

const lineItem = {
  image: {
    src:
      'https://cdn.shopify.com/s/files/1/1917/4769/products/Volc_-_orig.png?v=1492277039',
  },
  title: 'Volcano Vaporizer with Solid Valve',
  variant_title: 'Default Title',
  quantity: 1,
  line_price: '624.95',
};

describe('<CartItem />', () => {
  it("shouldn't throw error if no props are specified", () => {
    try {
      shallow(<CartItem />);
    } catch (error) {
      expect(error).toBeUndefined();
    }
  });

  it('decreaseQuantity should be fired when appropriate button input has been clicked', () => {
    const spy = jest.fn();
    const renderedComponent = mount(
      <CartItem lineItem={lineItem} decreaseQuantity={spy} />
    );
    renderedComponent
      .find('button')
      .at(0)
      .simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('increaseQuantity should be fired when appropriate button input has been clicked', () => {
    const spy = jest.fn();
    const renderedComponent = mount(
      <CartItem lineItem={lineItem} increaseQuantity={spy} />
    );
    renderedComponent
      .find('button')
      .at(1)
      .simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('renders data if provided', () => {
    const component = create(<CartItem lineItem={lineItem} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
