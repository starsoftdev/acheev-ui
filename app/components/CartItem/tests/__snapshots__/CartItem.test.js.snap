// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> renders data if provided 1`] = `
<div
  className="cartItem row mb-xl"
>
  <div
    className="small-3 column npl"
  >
    <img
      alt="Volcano Vaporizer with Solid Valve"
      className="cartItem__img"
      src="https://cdn.shopify.com/s/files/1/1917/4769/products/Volc_-_orig.png?v=1492277039"
    />
  </div>
  <div
    className="small-9 column"
  >
    <div
      className="cartItem__title mb-mn"
    >
      Volcano Vaporizer with Solid Valve
    </div>
    <div
      className="row align-middle mb-md fs-tn"
    >
      <div
        className="cartItem__quantityLabel shrink column"
      >
        QTY
      </div>
      <button
        className="cartItem__quantityChange shrink column"
        onClick={[Function]}
      >
        <span>
          -
        </span>
      </button>
      <div
        className="cartItem__quantityValue shrink column"
      >
        1
      </div>
      <button
        className="cartItem__quantityChange shrink column"
        onClick={[Function]}
      >
        <span>
          +
        </span>
      </button>
    </div>
    <div
      className="cartItem__price"
    >
      $
      624.95
    </div>
  </div>
</div>
`;
